.buildkite/scripts/release/step_update_version.sh

```diff
@@ -0,0 +1,151 @@
+git_remote_name="origin"
+git_branch="${BUILDKITE_BRANCH}"
+
+# End configuration
+
+if [[ -z "${git_branch}" ]]; then
+  >&2 echo "BUILDKITE_BRANCH is not set. This usually means you're trying to execute this script from the outside of Buildkite pipeline which is unsupported."
+  exit 1
+fi
+
+##
+# Check release type (prerelease or regular release)
+##
+
+release_type="prerelease"
+if [[ "${RELEASE_TYPE}" == "release" ]]; then
+  release_type="release"
+fi
+
+##
+# Check version number
+##
+
+if [[ "${release_type}" == "release" ]] && [[ -z "${RELEASE_VERSION}" ]]; then
+  >&2 echo "RELEASE_VERSION must be set when RELEASE_TYPE is 'release'"
+  exit 1
+fi
+
+##
+# Print run's configuration
+##
+
+echo "+++ Updating version on branch ${git_branch}"
+echo "release type: ${release_type}"
+echo "release version: ${RELEASE_VERSION}"
+echo "npm version prerelease prefix: ${npm_version_prerelease_prefix}"
+echo "git branch: ${git_branch}"
+echo "git push flags: ${git_push_flags}"
+echo "git remote URL: $(git remote get-url "${git_remote_name}")"
+
+##
+# Update package.json version string
+##
+
+echo "+++ Updating @elastic/eui version string"
+
+npm_version_args=(
+  --git-tag-version=false # disable tagging the new version
+  --sign-git-tag=false # disable signing the git tag
+  --commit-hooks=false # disable all git commit hooks
+)
+
+new_version=""
+if [[ "${release_type}" == "release" ]]; then
+  new_version=$(npm version "${npm_version_args[@]}" "${RELEASE_VERSION}")
+else
+  new_version=$(npm version "${npm_version_args[@]}" --preid=${npm_version_prerelease_prefix} prerelease)
+fi
+
+# npm version output prefixes the new version with 'v'
+new_version="${new_version:1}"
+
+echo "Updated @elastic/eui version string to ${new_version}"
+
+##
+# Check if version isn't already published if RELEASE_VERSION is set
+##
+
+echo "+++ :npm: Checking npm registry"
+if [[ "$(npm show @elastic/eui versions --json | jq 'index("${new_version}")')" != "null" ]]; then
+  >&2 echo "Version ${new_version} has already been published to npm and can't be overridden:"
+  >&2 echo "https://www.npmjs.com/package/@elastic/eui/v/${new_version}"
+  exit 2
+fi
+
+echo "Version ${new_version} hasn't been published to npm yet"
+
+##
+# Commit package.json
+##
+
+echo "+++ :git: Committing the version update"
+
+echo "Fetching OIDC token to sign the commit"
+SIGSTORE_ID_TOKEN="$(buildkite-agent oidc request-token --audience sigstore)"
+
+github_user_vault="secret/ci/elastic-eui/github_machine_user"
+
+git config --local user.name "$(vault read -field=name "${github_user_vault}")"
+git config --local user.email "$(vault read -field=email "${github_user_vault}")"
+git config --local commit.gpgsign true
+git config --local tag.gpgsign true
+git config --local gpg.x509.program gitsign
+git config --local gpg.format x509
+
+echo "Adding and committing package.json"
+git add package.json
+git commit -m "release: @elastic/eui v${new_version} [skip-ci]"
```

Comment:
Does this line need the `-S` flag to sign the commit?

```suggestion
git commit -S -m "release: @elastic/eui v${new_version} [skip-ci]"
```

Reply:
It doesn't since it's enabled by default on `git config commit.gpgsign true` line above